main/src/library/Map.flix

```diff
@@ -115,17 +117,10 @@ namespace Map {
     /// Otherwise, updates `m` with `k -> v`.
     ///
     def insertWithKey[k, v](f: (k, v, v) -> v, k: k, v: v, m: Map[k, v]): Map[k, v] =
```

Comment:
I changed the `Map.insert` functions to insert a new key (and its corresponding value) at the beginning of the underlying `List[(k, v)]`.  This is consistent with `Set` now (new elements in a `Set` are put at the beginning of the underlying `List`.

So the order we are maintaining with `Set` and `Map` for functions that require order (`fold`, `reduce`...) is most recently inserted element (`Set`) or key (`Map`) to least recently inserted element (`Set`) or key (`Map`).

Can I go ahead and note this in the documentation for these functions?

This brings up the following question:
If an existing key's value is updated in a `Map` or an element is added to a `Set` that already exists in the `Set`, should the corresponding mapping/element be moved to the front of the underlying list of the `Map`/`Set`?